File: DiaryNutrition/app/src/main/java/ediger/diarynutrition/data/source/dao/WeightDao.java
package ediger.diarynutrition.data.source.dao;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import ediger.diarynutrition.data.source.entities.Weight;

@Dao
public interface WeightDao {

    @Query("SELECT * FROM weight ORDER BY datetime DESC")
    LiveData<List<Weight>> loadWeight();

    /**
     * Get all records of weight for specific interval
     * @param from beginning of the period in milliseconds
     * @param to end of the period in milliseconds
     */
    @Query("SELECT * FROM weight WHERE datetime BETWEEN :from AND :to ORDER BY datetime ASC")
    LiveData<List<Weight>> getWeight(long from, long to);

    @Query("SELECT * FROM weight ORDER BY datetime DESC LIMIT 1")
    Weight getLastWeight();

    @Query("SELECT * FROM weight WHERE datetime BETWEEN :from AND :to")
    Weight getWeightForDay(long from, long to);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertWeight(Weight weight);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateWeight(Weight weight);

    @Query("DELETE FROM weight WHERE id = :id")
    int deleteWeightById(int id);

}

File: DiaryNutrition/app/src/main/java/ediger/diarynutrition/data/source/dao/WeightDao.kt
package ediger.diarynutrition.data.source.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import ediger.diarynutrition.data.source.entities.Weight

@Dao
interface WeightDao {
    @Query("SELECT * FROM weight ORDER BY datetime DESC")
    fun loadWeight(): LiveData<List<Weight>>

    /**
     * Get all records of weight for specific interval
     * @param from beginning of the period in milliseconds
     * @param to end of the period in milliseconds
     */
    @Query("SELECT * FROM weight WHERE datetime BETWEEN :from AND :to ORDER BY datetime ASC")
    fun getWeight(from: Long, to: Long): LiveData<List<Weight>>

    @Query("SELECT * FROM weight")
    suspend fun getWeight(): List<Weight>

    @Query("SELECT * FROM weight ORDER BY datetime DESC LIMIT 1")
    fun lastWeight(): Weight

    @Query("SELECT * FROM weight WHERE datetime BETWEEN :from AND :to")
    suspend fun getWeightForDay(from: Long, to: Long): Weight?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertWeight(weight: Weight)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun populateWeight(weights: List<Weight>)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun updateWeight(weight: Weight)

    @Query("DELETE FROM weight WHERE id = :id")
    suspend fun deleteWeightById(id: Int): Int
}